src/io/iohk/decco/test/snappy/SnappySpec.scala

```diff
@@ -0,0 +1,25 @@
+package io.iohk.decco.snappy
+
+import io.iohk.decco.test.utils.CodecTestingHelpers._
+import io.iohk.decco.definitions._
+import io.iohk.decco.snappy.auto._
+import org.scalatest.{FlatSpec, Matchers}
+import io.iohk.decco.BufferInstantiator.global.HeapByteBuffer
+
+class SnappySpec extends FlatSpec with Matchers {
+  behavior of "snappy"
+
+  it should "encode data properly" in {
+    testCodec[Array[Byte]]
+    testCodec[String]
+  }
+
+  it should "compress message" in {
```

Comment:
do we have an expected compressed output for a specific input?
It may be good to add at least one of these checks to verify the the correct compression algorithm is in use.

Reply:
(of course this is not needed for this PR, but caught my attention)

Reply:
Example would have to be generated manually as snappy codec for `Array[Byte]` doesn't just compress the content, but prepends it with size, once before compression and once after compression.